collector/thermalzone.go

```diff
@@ -0,0 +1,105 @@
+package collector
+
+import (
+	"github.com/StackExchange/wmi"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+func init() {
+	Factories["thermalzone"] = NewThermalZoneCollector
+}
+
+// A ThermalZoneCollector is a Prometheus collector for WMI Win32_PerfRawData_Counters_ThermalZoneInformation metrics
+type ThermalZoneCollector struct {
+	HighPrecisionTemperature *prometheus.Desc
+	PercentPassiveLimit      *prometheus.Desc
+	Temperature              *prometheus.Desc
+	ThrottleReasons          *prometheus.Desc
+}
+
+// NewThermalZoneCollector ...
+func NewThermalZoneCollector() (Collector, error) {
+	const subsystem = "thermalzone"
+	return &ThermalZoneCollector{
+		HighPrecisionTemperature: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "high_precision_temperature"),
+			"(HighPrecisionTemperature)",
+			nil,
+			nil,
+		),
+		PercentPassiveLimit: prometheus.NewDesc(
+			prometheus.BuildFQName(Namespace, subsystem, "percent_passive_limit"),
```

Comment:
Do you know what this one means? It is alway 100 on my machine.

Reply:
That's the % limit that the thermal zone is placing on the devices it controls. So 100% would be unconstrained

Reply:
Ah, ok. Putting that in the help text would be great :+1: